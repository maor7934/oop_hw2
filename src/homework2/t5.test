#check cross children:

CreateNode a1 10
CreateNode a2 10
CreateNode b1 10
CreateNode b2 10

CreateGraph A
CreateGraph B

AddNode A a1
AddNode A a2
AddNode B b1
AddNode B b2

AddEdge A a1 a2
AddEdge B b1 b2

ListNodes A
ListNodes B

ListChildren A a1
ListChildren A b1

AddEdge A a1 b1
ListChildren A a1
FindPath A a1 -> b2

AddNode A b1
ListChildren A a1
FindPath A a1 -> b2

AddEdge A a1 b1
ListChildren A a1
FindPath A a1 -> b2

AddNode A b2
ListChildren A a1
FindPath A a1 -> b2

AddEdge A b1 b2
ListChildren A a1
FindPath A a1 -> b2
