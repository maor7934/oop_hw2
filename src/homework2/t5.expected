#check cross children:

created node a1 with cost 10
created node a2 with cost 10
created node b1 with cost 10
created node b2 with cost 10

created graph A
created graph B

added node a1 to A
added node a2 to A
added node b1 to B
added node b2 to B

added edge from a1 to a2 in A
added edge from b1 to b2 in B

A contains: a1 a2
B contains: b1 b2

the children of a1 in A are: a2
Exception: homework2.CommandException: Graph A doesn't have node b1

Exception: homework2.CommandException: Graph A doesn't have node a1 \ b1
the children of a1 in A are: a2
no path found in A

added node b1 to A
the children of a1 in A are: a2
no path found in A

added edge from a1 to b1 in A
the children of a1 in A are: a2 b1
no path found in A

added node b2 to A
the children of a1 in A are: a2 b1
no path found in A

added edge from b1 to b2 in A
the children of a1 in A are: a2 b1
shortest path in A: a1 b1 b2
